gemstone/src/swapper/chainflip/provider.rs

```diff
@@ -46,29 +46,51 @@ impl ChainflipProvider {
         }
     }
 
-    fn get_boost_quote(&self, quote_response: &QuoteResponse) -> (String, u32, u32, Option<u32>, String) {
-        let egress_amount: String;
-        let slippage_bps: u32;
+    fn get_best_quote(mut quotes: Vec<QuoteResponse>, fee_bps: u32) -> (BigUint, u32, u32, ChainflipRouteData) {
+        quotes.sort_by(|a, b| b.egress_amount.cmp(&a.egress_amount));
+        let quote = &quotes[0];
+
+        let egress_amount: BigUint;
         let eta_in_seconds: u32;
+        let slippage_bps: u32;
         let boost_fee: Option<u32>;
         let estimated_price: String;
+        let dca_parameters: Option<DcaParameters>;
 
         // Use boost quote if available
-        if let Some(boost_quote) = &quote_response.boost_quote {
+        if let Some(boost_quote) = &quote.boost_quote {
             egress_amount = boost_quote.egress_amount.clone();
             slippage_bps = boost_quote.slippage_bps();
             eta_in_seconds = boost_quote.estimated_duration_seconds as u32;
             boost_fee = Some(boost_quote.estimated_boost_fee_bps);
             estimated_price = boost_quote.estimated_price.clone();
+            dca_parameters = boost_quote.dca_params.as_ref().map(|dca_params| DcaParameters {
+                number_of_chunks: dca_params.number_of_chunks,
+                chunk_interval: dca_params.chunk_interval_blocks,
+            });
         } else {
-            egress_amount = quote_response.egress_amount.clone();
-            slippage_bps = quote_response.slippage_bps();
-            eta_in_seconds = quote_response.estimated_duration_seconds as u32;
+            egress_amount = quote.egress_amount.clone();
+            slippage_bps = quote.slippage_bps();
+            eta_in_seconds = quote.estimated_duration_seconds as u32;
             boost_fee = None;
-            estimated_price = quote_response.estimated_price.clone();
+            estimated_price = quote.estimated_price.clone();
+            dca_parameters = quote.dca_params.as_ref().map(|dca_params| DcaParameters {
```

Comment:
My rust isn't that good, but will this code deal with regular quote when there are no dca_params present?

Reply:
yes, `dca_params` is an `Option` type and map only works when it has value, otherwise it will be `None`